File: packages/pigeon_example/android/src/main/kotlin/com/dsr_corporation/pigeon_example/DefaultPlatformInfoApi.kt
package com.dsr_corporation.pigeon_example

import PlatformInfo
import PlatformInfoApi
import VersionFlutterApi

// Android implementation for platform interface
class DefaultPlatformInfoApi(
    // Flutter interface
    private val api: VersionFlutterApi
) : PlatformInfoApi {

    // Get platform information
    override fun getPlatformInfo(callback: (Result<PlatformInfo>) -> Unit) {
        // Call flutter api to get app version
        api.getAppVersion { version ->
            version.onSuccess {
                callback(
                    Result.success(
                        PlatformInfo(
                            name = "Android",
                            version = android.os.Build.VERSION.RELEASE,
                            appVersion = it,
                        )
                    )
                )
            }.onFailure {
                callback(Result.failure(it))
            }
        }
    }
}

File: packages/pigeon_example/ios/Classes/DefaultPlatformInfoApi.swift
//
//  DefaultPlatformInfoApi.swift
//  Pods
//
//  Created by MJ on 12.07.2025.
//

// iOS implementation for platform interface
class DefaultPlatformInfoApi: PlatformInfoApi {

    // Flutter interface
    private let api: VersionFlutterApi

    init(api: VersionFlutterApi) {
        self.api = api
    }

    // Get platform information
    func getPlatformInfo(completion: @escaping (Result<PlatformInfo, Error>) -> Void) {
        // Call flutter api to get app version
        api.getAppVersion { result in
            switch result {
            case .success(let appVersion):
                completion(
                    .success(
                        PlatformInfo(
                            name: "iOS",
                            version: UIDevice.current.systemVersion,
                            appVersion: appVersion
                        )
                    )
                )
            case .failure(let error): completion(.failure(error))
            }
        }
    }
}
